proxmox/types/nodes/apt/repositories/errors.go

```diff
@@ -0,0 +1,38 @@
+/*
+	This Source Code Form is subject to the terms of the Mozilla Public
+	License, v. 2.0. If a copy of the MPL was not distributed with this
+	file, You can obtain one at https://mozilla.org/MPL/2.0/.
+*/
+
+package repositories
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/function"
+)
+
+//nolint:gochecknoglobals
+var (
+	// ErrCephVersionNameIllegal indicates an error for an illegal Ceph major version name.
+	ErrCephVersionNameIllegal = func(name string) error {
+		return function.NewFuncError(fmt.Sprintf("illegal Ceph major version name %q", name))
```

Comment:
I think `function.NewFuncError` is misused here. This part of the provider-supplied functions package, returning a specific error implementation. 
It also makes the API client code dependent on the hashicorp's code, that I'd prefer to avoid if possible.

I could probably just use the standard golang error wrapping technique, i.e.

```golang
var ErrStandardRepoHandleKindIllegal = errors.New("illegal APT standard repository handle kind")

...

	return StandardRepoHandleKindUnknown, fmt.Errorf(
		"parse APT standard repository handle kind %q: %w",
		input, ErrStandardRepoHandleKindIllegal,
	)
}
```